### ISSUE TITLE
Clang Power Tools mangles output for -Wmismatched-new-delete

### ISSUE THREAD

**Issue by Arminius** [Caphyon/clang-power-tools#388]
Take the following source code:
```
void f() {
  auto x = new int[10];
  delete x;
}

void g() {
  auto x = new int;
  delete[] x;
}
```

Using 'Compile (syntax only)' will cause the following text to appear in the output window:
```
array_delete.cpp(3): warning []'; did you mean 'delete[]'? [-Wmismatched-new-delete]: 'delete' applied to a pointer that was allocated with 'new
  delete x;
  ^
        []
array_delete.cpp(2): message []' here: allocated with 'new
  auto x = new int[10];
           ^
array_delete.cpp(8): warning []' applied to a pointer that was allocated with 'new'; did you mean 'delete'? [-Wmismatched-new-delete]: 'delete
  delete[] x;
  ^     ~~
array_delete.cpp(7): message: allocated with 'new' here
  auto x = new int;
           ^
2 warnings generated.
```
The square brackets are all over the place in these messages.

**Comment by hero101111** [comment]
Hello,

We've tested your code and the misalignment comes directly from Clang. [See pic](https://i.imgur.com/jjsgfS1.png).
Since this isn't a Clang Power Tools bug, there's nothing we can do about it. 
You should file a bug report directly to [LLVM](https://llvm.org/docs/HowToSubmitABug.html).

Regards,
Gabriel

**Comment by Arminius** [comment]
But the output in your pic is correct, whereas the output I have is not?

You have e.g.
```
'delete' applied to a pointer that was allocated with 'new[]'; did you mean 'delete[]'? [-Werror,-Wmismatched-new-delete]
```
whereas I have
```
warning []'; did you mean 'delete[]'? [-Wmismatched-new-delete]: 'delete' applied to a pointer that was allocated with 'new
```
The original message has been split; the first part (starting at the colon and ending before the first opening square bracket) appears last, i.e. the part that should appear after "`warning` " appears after `[-Wmismatched-new-delete]`.

**Comment by hero101111** [comment]
Apologies for the confusion.
We'll investigate further and update this thread when progress has been made.

**Comment by Enache-Ionut** [comment]
Hi, 

This issue was fixed but it is currently live only for the github repository.
The latest Visual Studio extension (v3.8) does not contain the fix. It will be included in the next release v3.9.

Thank you for your feedback!
